### PR TITLE
[SERVICE-494] Remove TrayIconClickEvent type definition

### PR DESCRIPTION
**Pull request by ryanpetersonOF**
Updates `@types/openfin` and removes our custom `TrayIconClicked` implementation and replaces it with the openfin type.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "@types/moment": "^2.13.0",
     "@types/node": "^9.4.6",
     "@types/node-fetch": "^2.3.4",
-    "@types/openfin": "^39.0.1",
+    "@types/openfin": "^43.0.1",
```
Comment (tomrobertsOF) [inline package.json]: Should we be concerned about using a version of the types intended for a newer runtime that we are currently running?

Comment (ryanpetersonOF) [reply package.json]: Blocked by SERVICE-583? 😀

Comment (bryangaleOF) [reply package.json]: Feel a bit concerned, but the runtime update is coming soon in any case, and if it compiles without major changes and works... 😬

Comment (tomrobertsOF) [reply package.json]: Yea, I don't expect it to actually cause any problems, just prefer being consistent about versioning for these things. It's pretty unlikely that the runtime upgrade won't be merged before the next release, so I guess we can just go ahead with this.

**Pull request merged**